.github/workflows/build-test-release.yml

```diff
@@ -0,0 +1,157 @@
+name: build, test and publish bs-emotion/bs-emotion-ppx
+
+on:  
+  push:    
+    branches: 
+      - master
+        
+  release:
+    types: [published]
```

Comment:
What does this setting do?

Reply:
It means when you do a github release, i.e. in published state, the pipeline is trigged. https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#onevent_nametypes